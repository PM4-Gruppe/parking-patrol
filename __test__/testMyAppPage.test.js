import renderer from 'react-test-renderer';
import MyApp from '../src/pages/_app';
import { useRouter } from 'next/router';
import { mocked } from 'jest-mock'

jest.mock('next/router', () => ({
    useRouter: jest.fn(),
}));

jest.mock('../src/components', () => 'Layout')
jest.mock('@auth0/nextjs-auth0/client', () => ({
    UserProvider: ({ children }) => <div>{children}</div>,
}));

jest.mock('../src/lib/apollo', () => ({
    __esModule: true,
    default: {
        query: () => {},
    },
}))

jest.mock('next/link', () => {
    const Link = ({ children, ...props }) => (
        <a {...props}>{children}</a>
    );
    Link.displayName = 'Link';
    return Link;
});

describe('MyApp', () => {
    beforeEach(() => {
        mocked(useRouter).mockReturnValue({
            back: jest.fn(),
        })
    });

    afterEach(() => {
        jest.resetAllMocks()
    });

    it('renders correctly', () => {
        const tree = renderer.create(<MyApp Component={() => <div>Test Component</div>} pageProps={{}} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})