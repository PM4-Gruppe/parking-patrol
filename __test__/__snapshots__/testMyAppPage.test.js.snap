// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyApp renders correctly 1`] = `
<div
  className="h-screen bg-gradient-to-t from-neutral-900 to-neutral-700"
>
  <div
    className="bg-neutral-800 p-5"
  >
    <nav
      className="md:ml-auto flex flex-wrap items-center text-base justify-center"
    >
      <a
        className="inline-flex items-center bg-gray-100 border-0 py-1 px-3 focus:outline-none hover:bg-gray-200 rounded text-base mt-4 md:mt-0"
        href="/api/auth/login"
      >
        Login
      </a>
    </nav>
  </div>
  <div>
    Test Component
  </div>
</div>
`;
